Composer/packages/client/src/pages/publish/index.tsx

```diff
@@ -0,0 +1,242 @@
+          iconName: 'CloudUpload',
+        },
+        onClick: () => selectPublishTarget(),
+      },
+      align: 'left',
+      dataTestid: 'publishPage-ToolBar-Publish',
+      disabled: false,
+    },
+  ];
+
+  useEffect(() => {
+    // load up the list of all publish targets
+    actions.getPublishTargetTypes();
+    if (settings.publishTargets) {
+      setPublishTarget(settings.publishTargets);
+    }
+  }, [settings]);
+
+  useEffect(() => {
+    // get selected target publish history
+    if (selectedTarget) {
+      actions.getPublishHistory(projectId, selectedTarget);
+    }
+  }, [selectedTarget]);
+
+  useEffect(() => {
+    setDialogProps({
+      title: 'Add a publish target',
+      type: DialogType.normal,
+      children: (
+        <CreatePublishTarget
+          targetTypes={publishTypes.map(type => {
+            return { key: type, text: type };
+          })}
+          onSave={savePublishTarget}
+          onCancel={closeAddDialog}
+        />
+      ),
+    });
+  }, [publishTypes]);
+
+  const addPublishTarget = () => {
+    setAddDialogHidden(false);
+  };
+
+  const selectPublishTarget = () => {
+    setPublishDialogHidden(false);
+  };
+
+  const closeAddDialog = () => {
+    setAddDialogHidden(true);
+  };
+
+  const closePublishDialog = () => {
+    setPublishDialogHidden(true);
+  };
+
+  const savePublishTarget = (name, type, configuration) => {
+    console.log(`save ${name} ${type} ${configuration}`);
+    actions.setSettings(
+      projectId,
+      botName,
+      {
+        ...settings,
+        publishTargets: [
+          {
+            name,
+            type,
+            configuration,
+          },
+        ].concat(publishTarget),
+      },
+      undefined
+    );
+  };
+
+  const publish = async comment => {
+    // publish to remote
+    if (selectedTarget) {
+      const sensitiveSettings = settingsStorage.get(botName);
+      await actions.publishToTarget(projectId, { ...selectedTarget, sensitiveSettings });
+    }
+  };
+
+  return (
+    <div>
+      <Dialog hidden={addDialogHidden} onDismiss={closeAddDialog} dialogContentProps={dialogProps}>
+        {dialogProps.children}
+      </Dialog>
+      <PublishDialog
+        hidden={publishDialogHidden}
+        onDismiss={closePublishDialog}
+        onSubmit={publish}
+        target={selectedTarget}
+      />
+      <ToolBar toolbarItems={toolbarItems} />
+      <div css={ContentHeaderStyle}>
+        <h1 css={HeaderText}>{formatMessage('Publish Profiles')}</h1>
+      </div>
+      <div css={ContentStyle} data-testid="Publish">
+        <div css={projectContainer}>
+          {publishTarget.length > 0 ? (
+            <Fragment>
+              <div css={targetListTiTle}>All profiles</div>
+              <TargetList list={settings.publishTargets} onSelect={setSelectedTarget} selectedTarget={selectedTarget} />
+            </Fragment>
+          ) : (
+            <div>This bot does not have a profile to publish yet</div>
```

Comment:
This div is currently unstyled.  See design.

Reply:
change it